rt_bene_model_training/pytorch/rtbene_dataset.py

```diff
@@ -0,0 +1,72 @@
+import os
+
+import numpy as np
+from PIL import Image
+from torch.utils import data
+from torchvision import transforms
+from tqdm import tqdm
+
+
+class RTBENEH5Dataset(data.Dataset):
+
+    def __init__(self, h5_file, subject_list=None, transform=None, loader_desc="train"):
+        self._h5_file = h5_file
+        self._transform = transform
+        self._subject_labels = []
+        self._positive_labels = 0
+        self._total_labels = 0
+
+        assert subject_list is not None, "Must pass a list of subjects to load the data for"
+
+        if self._transform is None:
+            self._transform = transforms.Compose([transforms.Resize((224, 224), Image.BICUBIC),
+                                                  transforms.ToTensor(),
+                                                  transforms.Normalize(mean=[0.485, 0.456, 0.406],
+                                                                       std=[0.229, 0.224, 0.225])])
+
+        _wanted_subjects = ["s{:03d}".format(_i) for _i in subject_list]
+
+        for grp_s_n in tqdm(_wanted_subjects, desc="Loading ({}) subject metadata...".format(loader_desc)):  # subjects
+            for grp_i_n, grp_i in h5_file[grp_s_n].items():  # images
+                if "image" in grp_i.keys() and "label" in grp_i.keys():
```

Comment:
Is there ever a case where "image" and "label" are not in the keys?

Reply:
not in RT-BENE dataset, similar to the training code for RT-GENE, I was hoping to future-proof it if there was ever another dataset out there that we can use.

Reply:
Gotcha